### PR TITLE
EVG-15092: convert platform into OS and architecture

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-15092

### Description 
Split the platform into OS and architecture, which will help us in choosing which agent to download on the host. It's also necessary to know the OS to assign the pod to the correct cluster.

I also fixed some minor issues in the REST route to create pods.

### Testing 
Normal REST and DB model tests.

### CODE REVIEW

**Review by ybrill — APPROVED** [review]
all my comments are on code that's just being moved around

File: rest/route/pod.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		p:  model.APICreatePod{},
+	}
+}
+
+// Parse fetches the podID and JSON payload from the HTTP request.
+func (h *podPostHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+
+	b, err := ioutil.ReadAll(body)
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply rest/route/pod.go]: Sure, I changed it.

File: rest/route/pod.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	if err := json.Unmarshal(b, &h.p); err != nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("API error while unmarshalling JSON"),
+		}
+	}
+
+	if utility.FromStringPtr(h.p.Image) == "" {
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply rest/route/pod.go]: Done.

File: rest/route/pod.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+// Run creates a new resource based on the Request-URI and JSON payload.
+func (h *podPostHandler) Run(ctx context.Context) gimlet.Responder {
+	res, err := h.sc.CreatePod(h.p)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Creating new pod"))
+	}
+
+	responder := gimlet.NewJSONResponse(res)
+
+	if err := responder.SetStatus(http.StatusCreated); err != nil {
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply rest/route/pod.go]: Done.

**Review by ybrill — APPROVED** [review]

**Pull request merged**